src/invidious/yt_backend/extractors.cr

```diff
@@ -656,27 +660,34 @@ private module Parsers
 
       thumbnail = thumbnail_view_model.dig("image", "sources", 0, "url").as_s
 
-      # This complicated sequences tries to extract the following data structure:
-      # "overlays": [{
-      #   "thumbnailOverlayBadgeViewModel": {
-      #     "thumbnailBadges": [{
-      #       "thumbnailBadgeViewModel": {
-      #         "text": "430 episodes",
-      #         "badgeStyle": "THUMBNAIL_OVERLAY_BADGE_STYLE_DEFAULT"
-      #       }
-      #     }]
-      #   }
-      # }]
-      #
-      # NOTE: this simplistic `.to_i` conversion might not work on larger
-      # playlists and hasn't been tested.
-      video_count = thumbnail_view_model.dig("overlays").as_a
+      # Parse overlays badges
+      overlays = thumbnail_view_model.dig("overlays").as_a
         .compact_map(&.dig?("thumbnailOverlayBadgeViewModel", "thumbnailBadges").try &.as_a)
         .flatten
-        .find(nil, &.dig?("thumbnailBadgeViewModel", "text").try { |node|
-          {"episodes", "videos"}.any? { |str| node.as_s.ends_with?(str) }
-        })
-        .try &.dig("thumbnailBadgeViewModel", "text").as_s.to_i(strict: false)
+
+      # Detect Mix playlists via icon imageName at any index or text case-insensitively
+      is_mix = overlays.any? { |badge|
+        sources = badge.dig?("thumbnailBadgeViewModel", "icon", "sources").try &.as_a || [] of JSON::Any
+        has_mix_icon = sources.any? { |s| s.dig?("clientResource", "imageName").try &.as_s == "MIX" }
+        text = badge.dig?("thumbnailBadgeViewModel", "text").try &.as_s || ""
+        has_mix_text = text.downcase == "mix"
+        has_mix_icon || has_mix_text
+      }
+
+      # Fallback: RD-prefixed playlist IDs are Mixes
+      is_mix ||= playlist_id.starts_with? "RD"
```

Comment:
I'm not sure to fully understand why you need all that extra logic to determine if the playlist is a mix, when that last check seems robust enough? (as far as I know, mixes have always began with `RD`).